src/components/ebay-filter-menu/index.js

```diff
@@ -0,0 +1,80 @@
+const assign = require('core-js-pure/features/object/assign');
+const scrollKeyPreventer = require('makeup-prevent-scroll-keys');
+const rovingTabindex = require('makeup-roving-tabindex');
+const eventUtils = require('../../common/event-utils');
+const template = require('./template.marko');
+
+module.exports = require('marko-widgets').defineComponent({
+    template,
+    getInitialState(input) {
+        return assign({}, input, {
+            items: (input.items || []).map(item => assign({}, item))
+        });
+    },
+    onRender(event) {
+        this.contentEl = this.el.querySelector('.filter-menu__menu, .filter-menu-button__menu');
+
+        if (this.state.withButton) {
+            this.contentEl = this.el;
+        }
+
+        if (event.firstRender) {
+            if (this.state.variant !== 'form') {
+                // FIXME: should be outside of firstRender, but only when itemEls changes
```

Comment:
This is going to cause hard to track bugs, would be good to get this `FIXME` fixed!

Reply:
Where should we be doing this for Marko 3 components?

Reply:
On every render (not just the first) we should be setting up the rovingTabIndex, then `onBeforeUpdate` we should destroy it.

Reply:
Cool. Let me make those updates.